### ISSUE TITLE
sdhc-insert/ remove jobs not working for USB CARDREADER component

### ISSUE THREAD

**Issue by scotthu27** [canonical/checkbox#1456]
### Bug Description

The SD card reader in some PC platforms is a combo card reader combines USB & SDHC.
```
path: /devices/pci0000:00/0000:00:14.0/usb1/1-7/1-7:1.0/host0/target0:0:0/0:0:0:0
name: sda
bus: scsi
category: CARDREADER
driver: sd
product: SD/MMC/MS PRO
vendor: Generic-
product_slug: SD_MMC_MS_PRO
vendor_slug: Generic-
```

The case not working since [Removed udisks2 dependencies from mediacard](https://github.com/canonical/checkbox/pull/1291/commits/edb595a081e91365276c58905d9fbcb39cd91590) changed

### To Reproduce

1. checkbox-cli run com.canonical.certification::mediacard/sdhc-insert

### Environment

Checkbox Type: Debian
Checkbox Version: 4.2.0~dev59~ubuntu24.04.1

### Relevant log output

```shell
$ /usr/bin/checkbox-support-run_watcher insertion mediacard


INSERT NOW


Timeout: 30 seconds
Traceback (most recent call last):
  File "/usr/bin/checkbox-support-run_watcher", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3/dist-packages/checkbox_support/helpers/timeout.py", line 144, in _f
    return run_with_timeout(f, timeout_s, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3/dist-packages/checkbox_support/helpers/timeout.py", line 114, in run_with_timeout
    raise TimeoutError("Task unable to finish in {}s".format(timeout_s))
TimeoutError: Task unable to finish in 30s


$ /usr/bin/checkbox-support-run_watcher insertion usb2


INSERT NOW


Timeout: 30 seconds
high_speed_usb was inserted. Controller: xhci_hcd, Number: 15
usable partition: sda1
USB2 insertion test passed.
no env var PLAINBOX_SESSION_SHARE


ubuntu@localhost:~$ lsblk
NAME        MAJ:MIN RM   SIZE RO TYPE MOUNTPOINTS
loop0         7:0    0     4K  1 loop /snap/bare/5
loop1         7:1    0  32.7M  1 loop /snap/bugit/34
loop2         7:2    0 104.2M  1 loop /snap/core/17200
loop3         7:3    0  39.5M  1 loop /snap/bugit/54
loop4         7:4    0  74.3M  1 loop /snap/core22/1564
loop5         7:5    0  74.3M  1 loop /snap/core22/1586
loop6         7:6    0 269.8M  1 loop /snap/firefox/4793
loop7         7:7    0 505.1M  1 loop /snap/gnome-42-2204/176
loop8         7:8    0  10.5M  1 loop /snap/snap-store/1173
loop9         7:9    0    17M  1 loop /snap/factory-reset-tools/40
loop10        7:10   0  91.7M  1 loop /snap/gtk-common-themes/1535
loop11        7:11   0 271.2M  1 loop /snap/firefox/4848
loop12        7:12   0  10.7M  1 loop /snap/firmware-updater/127
loop13        7:13   0  38.8M  1 loop /snap/snapd/21759
loop14        7:14   0   476K  1 loop /snap/snapd-desktop-integration/157
loop15        7:15   0   500K  1 loop /snap/snapd-desktop-integration/178
loop16        7:16   0 149.6M  1 loop /snap/thunderbird/497
loop17        7:17   0 149.6M  1 loop /snap/thunderbird/507
sda           8:0    1  14.8G  0 disk 
└─sda1        8:1    1  14.8G  0 part /media/ubuntu/3634-6530
nvme0n1     259:0    0 953.9G  0 disk 
├─nvme0n1p1 259:1    0     1G  0 part /boot/efi
├─nvme0n1p2 259:2    0    12G  0 part 
└─nvme0n1p3 259:3    0 940.8G  0 part /
```

**Comment by pieqq** [comment]
For the devices having issues, can you:

- provide their CID
- provide the output of `udevadm info -e` so that we have the udev database we can add to the checkbox support unit tests and prevent regressions later on

**Comment by scotthu27** [comment]
CID: [202408-34238](https://certification.canonical.com/hardware/202408-34238/)	SEN16MLK-DVT1-C1M
[udevadm.txt](https://github.com/user-attachments/files/16904733/udevadm.txt)

**Comment by scotthu27** [comment]
```
$ journalctl -f
Sep 06 08:28:58 localhost wpa_supplicant[985]: wlp2s0: CTRL-EVENT-SIGNAL-CHANGE above=1 signal=-39 noise=9999 txrate=458800
Sep 06 08:29:08 localhost systemd[1]: NetworkManager-dispatcher.service: Deactivated successfully.
Sep 06 08:29:08 localhost sshd[7742]: Accepted publickey for ubuntu from 10.102.142.189 port 57168 ssh2: RSA SHA256:8oAdf+cF6qp66hAUgh9tsza5VBROYTV8XxSaBfKW4aI
Sep 06 08:29:08 localhost sshd[7742]: pam_unix(sshd:session): session opened for user ubuntu(uid=1000) by ubuntu(uid=0)
Sep 06 08:29:08 localhost systemd-logind[947]: New session 43 of user ubuntu.
Sep 06 08:29:08 localhost systemd[1]: Started session-43.scope - Session 43 of User ubuntu.
Sep 06 08:29:28 localhost systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Sep 06 08:30:00 localhost systemd[1]: Starting sysstat-collect.service - system activity accounting tool...
Sep 06 08:30:00 localhost systemd[1]: sysstat-collect.service: Deactivated successfully.
Sep 06 08:30:00 localhost systemd[1]: Finished sysstat-collect.service - system activity accounting tool.
Sep 06 08:30:01 localhost CRON[7804]: pam_unix(cron:session): session opened for user root(uid=0) by root(uid=0)
Sep 06 08:30:01 localhost CRON[7805]: (root) CMD ([ -x /etc/init.d/anacron ] && if [ ! -d /run/systemd/system ]; then /usr/sbin/invoke-rc.d anacron start >/dev/null; fi)
Sep 06 08:30:01 localhost CRON[7804]: pam_unix(cron:session): session closed for user root
Sep 06 08:30:11 localhost kernel: usb 1-7: new high-speed USB device number 18 using xhci_hcd
Sep 06 08:30:11 localhost kernel: usb 1-7: New USB device found, idVendor=0bda, idProduct=0177, bcdDevice=77.11
Sep 06 08:30:11 localhost kernel: usb 1-7: New USB device strings: Mfr=1, Product=2, SerialNumber=3
Sep 06 08:30:11 localhost kernel: usb 1-7: Product: USB2.0-CRW
Sep 06 08:30:11 localhost kernel: usb 1-7: Manufacturer: Generic
Sep 06 08:30:11 localhost kernel: usb 1-7: SerialNumber: 20121112761000000
Sep 06 08:30:11 localhost kernel: ums-realtek 1-7:1.0: USB Mass Storage device detected
Sep 06 08:30:11 localhost kernel: scsi host0: usb-storage 1-7:1.0
Sep 06 08:30:11 localhost mtp-probe[7815]: checking bus 1, device 18: "/sys/devices/pci0000:00/0000:00:14.0/usb1/1-7"
Sep 06 08:30:11 localhost mtp-probe[7815]: bus: 1, device: 18 was not an MTP device
Sep 06 08:30:11 localhost mtp-probe[7821]: checking bus 1, device 18: "/sys/devices/pci0000:00/0000:00:14.0/usb1/1-7"
Sep 06 08:30:11 localhost mtp-probe[7821]: bus: 1, device: 18 was not an MTP device
Sep 06 08:30:12 localhost kernel: scsi 0:0:0:0: Direct-Access     Generic- SD/MMC/MS PRO    1.00 PQ: 0 ANSI: 4
Sep 06 08:30:12 localhost kernel: sd 0:0:0:0: Attached scsi generic sg0 type 0
Sep 06 08:30:13 localhost kernel: sd 0:0:0:0: [sda] 31116288 512-byte logical blocks: (15.9 GB/14.8 GiB)
Sep 06 08:30:13 localhost kernel: sd 0:0:0:0: [sda] Write Protect is off
Sep 06 08:30:13 localhost kernel: sd 0:0:0:0: [sda] Mode Sense: 2f 00 00 00
Sep 06 08:30:13 localhost kernel: sd 0:0:0:0: [sda] Write cache: disabled, read cache: enabled, doesn't support DPO or FUA
Sep 06 08:30:13 localhost kernel:  sda: sda1
Sep 06 08:30:13 localhost kernel: sd 0:0:0:0: [sda] Attached SCSI removable disk
Sep 06 08:30:13 localhost (udev-worker)[7827]: sda: Process '/usr/bin/unshare -m /usr/bin/snap auto-import --mount=/dev/sda' failed with exit code 1.
Sep 06 08:30:13 localhost systemd[1]: anacron.service - Run anacron jobs was skipped because of an unmet condition check (ConditionACPower=true).
Sep 06 08:30:13 localhost (udev-worker)[7827]: sda1: Process '/usr/bin/unshare -m /usr/bin/snap auto-import --mount=/dev/sda1' failed with exit code 1.
Sep 06 08:30:13 localhost kernel: FAT-fs (sda1): Volume was not properly unmounted. Some data may be corrupt. Please run fsck.
Sep 06 08:30:13 localhost udisksd[950]: Mounted /dev/sda1 at /media/ubuntu/3634-6530 on behalf of uid 1000
Sep 06 08:30:13 localhost gnome-shell[1628]: GFileInfo created without standard::icon
Sep 06 08:30:13 localhost gnome-shell[1628]: file ../../../gio/gfileinfo.c: line 1765 (g_file_info_get_icon): should not be reached

```

**Comment by zongminl** [comment]
Fixed by #1438, #1477 and #1485
